src/main/java/docSharing/entities/document/File.java

```diff
@@ -0,0 +1,36 @@
+package docSharing.entities.document;
+
+import docSharing.entities.Permission;
+import docSharing.entities.User;
+
+import javax.persistence.*;
+import java.util.HashMap;
+import java.util.List;
+
+@Entity
+@Table(name="file")
+public class File {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private final int id;
+
+    @Column(unique=true)
+    private String url;
+
+    private HashMap<Permission, List<User>> authorized;
+    private MetaData metaData;
+    private final List<UpdateLog> updateLogs;
+
+    //empty constructor
+    public File(){
+        this.id=0;
+        this.updateLogs=null;
+    }
+    public File(int id, String url, HashMap<Permission, List<User>> authorized, MetaData metaData, List<UpdateLog> updateLogs) {
+        this.id = id;
```

Comment:
כמו בDocument, לא צריך לשלוח פה כלום חוץ מurl ו-User

Reply:
I changed to: user, parent, title, URL: 

public File(User user,File parent,String title,String url) {

URL will create by us (in the next sprint).

Reply:
לא צריך לשלוח id.
ואת כל האתחולים תכניסי לקונסטרקטור הריק ופשוט תקראי לו, כי את רוצה שהאתחולים ייקרו גם אם מאתחלים בקונסטרקטור הריק...
אם לא מאתחלים אף פעם עם הריק אז תעשי אותו private,
אבל בכל מקרה תכניסי לשם את כל האתחולים